src/subcommands/wallet.rs

```diff
@@ -293,42 +259,46 @@ impl<'a> WalletSubCommand<'a> {
         let mut lock_scripts = vec![(
             Script::from(&from_address_payload),
             sighash_placeholder_witness.clone(),
+            SinceSource::default(),
         )];
 
         let from_lock_arg = H160::from_slice(from_address.payload().args().as_ref()).unwrap();
         let mut path_map: HashMap<H160, DerivationPath> = Default::default();
-        let (change_address_payload, change_path) = if let Some(last_change_address) =
-            last_change_address_opt.as_ref()
-        {
-            // Behave like HD wallet
-            let change_last =
-                H160::from_slice(last_change_address.payload().args().as_ref()).unwrap();
-            let key_set = self.plugin_mgr.keystore_handler().derived_key_set(
-                from_lock_arg.clone(),
-                receiving_address_length,
-                change_last.clone(),
-                DERIVE_CHANGE_ADDRESS_MAX_LEN,
-                None,
-            )?;
-            let mut change_path_opt = None;
-            for (path, hash160) in key_set.external.iter().chain(key_set.change.iter()) {
-                if hash160 == &change_last {
-                    change_path_opt = Some(path.clone());
+        let (change_address_payload, change_path) =
+            if let Some(last_change_address) = last_change_address_opt.as_ref() {
+                // Behave like HD wallet
+                let change_last =
+                    H160::from_slice(last_change_address.payload().args().as_ref()).unwrap();
+                let key_set = self.plugin_mgr.keystore_handler().derived_key_set(
+                    from_lock_arg.clone(),
+                    receiving_address_length,
+                    change_last.clone(),
+                    DERIVE_CHANGE_ADDRESS_MAX_LEN,
+                    None,
+                )?;
+                let mut change_path_opt = None;
+                for (path, hash160) in key_set.external.iter().chain(key_set.change.iter()) {
```

Comment:
we may use owned iter to reduce some `hash160` and `path` clone:

```suggestion
                for (path, hash160) in key_set.external.into_iter().chain(key_set.change.into_iter()) {
```